src/sql/SqlStatement.ts

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (c) 2008-2021, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as Long from 'long';
+
+/**
+ * The expected statement result type.
+ */
+export enum SqlExpectedResultType {
+    /** The statement may produce either rows or an update count. */
+    ANY,
+    /** The statement must produce rows. An exception is thrown if the statement produces an update count. */
+    ROWS,
+    /** The statement must produce an update count. An exception is thrown if the statement produces rows. */
+    UPDATE_COUNT
+}
+
+export type SqlExpectedResultTypeStrings = keyof typeof SqlExpectedResultType;
+
+/**
+ * Options used to change behavior of an SQL query.
+ */
+export interface SqlStatementOptions {
+    /**
+     * The schema name. The engine will try to resolve the non-qualified object identifiers from the statement in the
+     * given schema. If not found, the default search path will be used, which looks for objects in the predefined
+     * schemas `partitioned` and `public`.
+     *
+     * * The schema name is case sensitive. For example, `foo` and `Foo` are different schemas.
+     * * The default value is `null` meaning only the default search path is used.
+     */
+    schema?: string | null;
+    /**
+     * The execution timeout in milliseconds. If the timeout is reached for a running statement, it will be cancelled forcefully.
+     * Zero value means no timeout. `-1` means that the timeout in server config will be used.
+     * Other negative values are prohibited.
+     * Defaults to `-1`.
+     */
+    timeoutMillis?: Long | number;
```

Comment:
BTW, thinking about this again and looking at the other config options we used throughout the client (like `maxBackoffMillis` or `clusterConnectTimeoutMillis` of `ConnectionRetryConfig`, which are `long`s in Java), I have a feeling that just allowing numbers here would be better, constructing a long is harder and the range of the numbers are huge enough. What do you think about it?

Reply:
I agree that can be addressed later, a backlog issue would be good.

Reply:
Created an issue https://github.com/hazelcast/hazelcast-nodejs-client/issues/912

Reply:
I meant removing the Long from the type and making it `timeoutMillis?: number;`. This cannot be changed later. 

I think this will simplify the API and implementation.